rust/processor/src/db/common/models/user_transactions_models/parquet_user_transactions.rs

```diff
@@ -0,0 +1,133 @@
+    utils::util::{get_entry_function_from_user_request, parse_timestamp, standardize_address},
+};
+use allocative::Allocative;
+use anyhow::Result;
+use aptos_protos::{
+    transaction::v1::{
+        TransactionInfo, UserTransaction as UserTransactionPB, UserTransactionRequest,
+    },
+    util::timestamp::Timestamp,
+};
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+#[derive(Allocative, Clone, Debug, Default, Deserialize, ParquetRecordWriter, Serialize)]
+pub struct UserTransaction {
+    pub txn_version: i64,
+    pub block_height: i64,
+    pub parent_signature_type: String,
+    pub sender: String,
+    pub sequence_number: i64,
+    pub max_gas_amount: u64,
+    pub expiration_timestamp_secs: u64,
+    pub gas_unit_price: u64,
+    pub gas_used: u64,
+    #[allocative(skip)]
+    pub timestamp: chrono::NaiveDateTime,
+    pub entry_function_id_str: String,
+    pub epoch: i64,
+    pub gas_fee_payer_address: Option<String>,
+    pub is_transaction_success: bool,
+    pub storage_refund_amount: u64,
+    pub num_signatures: i64,
+}
+
+impl NamedTable for UserTransaction {
+    const TABLE_NAME: &'static str = "user_transactions";
+}
+
+impl HasVersion for UserTransaction {
+    fn version(&self) -> i64 {
+        self.txn_version
+    }
+}
+
+impl GetTimeStamp for UserTransaction {
+    fn get_timestamp(&self) -> chrono::NaiveDateTime {
+        self.timestamp
+    }
+}
+
+impl UserTransaction {
+    pub fn from_transaction(
+        txn: &UserTransactionPB,
+        txn_info: &TransactionInfo,
+        fee_statement: Option<FeeStatement>,
+        timestamp: &Timestamp,
+        block_height: i64,
+        epoch: i64,
+        version: i64,
+    ) -> Self {
+        let user_request = txn
+            .request
+            .as_ref()
+            .expect("Sends is not present in user txn");
+        let gas_fee_payer_address = match user_request.signature.as_ref() {
+            Some(signature) => Signature::get_fee_payer_address(signature, version),
+            None => None,
+        };
+        let num_signatures = Self::get_signatures(user_request, version, block_height).len() as i64;
+
+        Self {
+            txn_version: version,
+            block_height,
+            parent_signature_type: txn
+                .request
+                .as_ref()
+                .unwrap()
+                .signature
+                .as_ref()
+                .map(Signature::get_signature_type)
+                .unwrap_or_default(),
+            sender: standardize_address(&user_request.sender),
+            sequence_number: user_request.sequence_number as i64,
+            max_gas_amount: user_request.max_gas_amount,
+            expiration_timestamp_secs: user_request
+                .expiration_timestamp_secs
+                .as_ref()
+                .expect("Expiration timestamp is not present in user txn")
+                .seconds as u64,
```

Comment:
is u64 enough here?

Reply:
yea should be.

Reply:
b/c the source says u64 so it cannot exceed that.